x-pack/agent/pkg/agent/program/testdata/single_config.yml

```diff
@@ -26,18 +26,24 @@ datasources:
       streams:
         - metricset: status
           dataset: docker.status
+      hosts: ["http://127.0.0.1:8080"]
     - type: logs
       streams:
         - paths:
           - /var/log/hello1.log
           - /var/log/hello2.log
+      vars:
+        var: value
   - namespace: testing
     use_output: default
     inputs:
       - type: apache/metrics
         streams:
           - enabled: true
             metricset: info
+            hosts: ["http://apache.remote"]
```

Comment:
I wonder if this is just a bad example? Because the same field should not exist in stream an input.

Reply:
yes i just needed to test override functionality somehow

Reply:
Wasn't the conclusion we don't support override? @ph ?

Reply:
@ruflin @michalpristas I think the conclusion was effectively we do not support overrides, I've looked at our email and this was the behavior he had in mind.

- for keys defined in the input section we apply them in the streams.
- for keys defined in the input section and present in the streams we fail hard.
- If a key is not defined in the inputs but defined in the stream we take the value of the stream.

Reply:
@ruflin yes we don't this test is just to make sure nothing is replaced by anything defined at inputs level so we make sure it does not happen as we copy all the fields down the tree. where we need overrride are situation when we generate somethin in tree and we want to make sure it is not lost (e.g. processors) we generate on stream level (just like index) but then we merge it with top level processors (we cannot generate at top level as fields are stream specific)

Reply:
Based on the comment from @ph , I would assume this test should fail hard? "for keys defined in the input section and present in the streams we fail hard."

Reply:
i think you're right, will make it fail entirely